File: coding/src/main/scala/org/shev4ik/algorithms/sorting/InsertionSorting.scala
package org.shev4ik.algorithms.sorting

object InsertionSorting extends App {

  val arr = Array(6,3,2,4,1,5)

  for(index <- 1.until(arr.length)){

    val elem = arr(index)
    var insertIndex = index - 1
    while(insertIndex >= 0 && arr(insertIndex) > elem){
      arr(insertIndex+1) = arr(insertIndex)
      insertIndex -= 1
    }

    arr(insertIndex+1) = elem
  }

  println(arr.toList)
}

File: coding/src/main/kotlin/org.kotlin.algorithms/InsertionSorting.kt
package org.kotlin.algorithms

fun main(args: Array<String>){
    val arr = arrayOf(2,1,3,5,4)
    for(i in 1.until(arr.size)) {
        val elem = arr[i]
        var index = i - 1
        while(index >= 0 && elem < arr[index]){
            arr[index+1] = arr[index]
            index--
        }
        arr[index + 1] = elem
    }
    arr.forEach { i -> println(i) }
    doWorld()

}


suspend fun doWorld() = coroutineScope {
    launch {
        delay(1000L)
        println("World")
    }
    println("Hello")
}